src/core/Interpreter.ts

```diff
@@ -65,11 +65,17 @@ class Interpreter {
     chain: Chain
     db: DatabaseInterface
 
-    constructor(chain: Chain, userAddress: string | null = null, db: DatabaseInterface | null = null) {
+    constructor(chain: Chain, userAddress: string | null = null, db: DatabaseInterface | null = null, additionalContractInterpreters?: Record<string, InterpreterMap>) {
         this.chain = chain
         this.userAddress = (userAddress && AddressZ.parse(userAddress)) || null
         this.db = db || new NullDatabaseInterface()
 
+        if (additionalContractInterpreters) {
+          for (const [address, map] of Object.entries(additionalContractInterpreters)) {
+              this.contractSpecificInterpreters[address.toLowerCase()] = map as InterpreterMap
+          }
+        }
+
         for (const [address, map] of Object.entries(contractInterpreters)) {
```

Comment:
do we wanna switch the orders of these two for loops so that the passed in maps get priority over the hardcoded defaults?